src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer.cs

```diff
@@ -356,93 +447,427 @@ bool IsCreateHelper(IMethodSymbol method)
+                    AnalyzeTitleCore, containingType, localizableTitles, resourceDataValueMap,
+                    localizableStringType, localizableResourceStringType);
             }
+        }
+
+        private static void AnalyzeTitleCore(string title, IArgumentOperation argumentOperation, Location fixLocation, Action<Diagnostic> reportDiagnostic)
+        {
+            var isMultiSentences = IsMultiSentences(title);
+            var endsWithPeriod = EndsWithPeriod(title);
+            var containsLineReturn = ContainsLineReturn(title);
+            if (isMultiSentences || endsWithPeriod || containsLineReturn)
+            {
+                var fixedTitle = endsWithPeriod ? RemoveTrailingPeriod(title) : title;
+                fixedTitle = isMultiSentences ? FixMultiSentences(fixedTitle) : fixedTitle;
+                fixedTitle = containsLineReturn ? FixLineReturns(fixedTitle, allowMultisentences: false) : fixedTitle;
+                Debug.Assert(title != fixedTitle);
 
-            operationAnalysisContext.ReportDiagnostic(creation.Value.CreateDiagnostic(UseLocalizableStringsInDescriptorRule, WellKnownTypeNames.MicrosoftCodeAnalysisLocalizableString));
+                ReportDefineDiagnosticArgumentCorrectlyDiagnostic(DefineDiagnosticTitleCorrectlyRule,
+                    argumentOperation, fixedTitle, fixLocation, reportDiagnostic);
+            }
+        }
 
-            if (!TryGetNonEmptyConstantStringValue(titleArgument, out var title))
+        private static void ReportDefineDiagnosticArgumentCorrectlyDiagnostic(
+            DiagnosticDescriptor descriptor,
+            IArgumentOperation argumentOperation,
+            string fixValue,
+            Location fixLocation,
+            Action<Diagnostic> reportDiagnostic)
+        {
+            // Additional location in an additional document does not seem to be preserved
+            // from analyzer to code fix due to a Roslyn bug.
+            // We workaround this bug by passing additional document file path and location span as strings.
+            // TODO: File a Roslyn bug.
+
+            var additionalLocations = ImmutableArray<Location>.Empty;
+            var properties = ImmutableDictionary<string, string>.Empty.Add(DefineDescriptorArgumentCorrectlyFixValue, fixValue);
+            if (fixLocation.IsInSource)
             {
-                return;
+                additionalLocations = additionalLocations.Add(fixLocation);
             }
+            else
+            {
+                var span = fixLocation.SourceSpan;
+                var locationInfo = $"{span.Start};{span.Length};{fixLocation.GetLineSpan().Path}";
+                properties = properties.Add(DefineDescriptorArgumentCorrectlyFixAdditionalDocumentLocationInfo, locationInfo);
+            }
+
+            var location = argumentOperation.Syntax.GetLocation();
+            var diagnostic = Diagnostic.Create(descriptor, location, additionalLocations: additionalLocations, properties: properties);
```

Comment:
Assuming the overload exists.

```suggestion
            var diagnostic = argumentOperation.CreateDiagnostic(descriptor, additionalLocations: additionalLocations, properties: properties);
```

Reply:
Yes, thanks. Will change.

Reply:
I added new overloads and made this change.